code/modules/mob/living/basic/jungle/venus_human_trap.dm

```diff
@@ -161,76 +177,73 @@
 	var/playable_plant = TRUE
 
-/mob/living/simple_animal/hostile/venus_human_trap/Life(seconds_per_tick = SSMOBS_DT, times_fired)
+/mob/living/basic/venus_human_trap/Initialize(mapload)
 	. = ..()
-	pull_vines()
+	AddElement(/datum/element/lifesteal, 5)
+	var/datum/action/cooldown/mob_cooldown/vine_tangle/tangle = new(src)
+	tangle.Grant(src)
+	ai_controller.set_blackboard_key(BB_HUMANTRAP_VINE_TANGLE, tangle)
 
-/mob/living/simple_animal/hostile/venus_human_trap/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+/mob/living/basic/venus_human_trap/Life(seconds_per_tick = SSMOBS_DT, times_fired)
 	. = ..()
-	pixel_x = base_pixel_x + (dir & (NORTH|WEST) ? 2 : -2)
+	if(!..())
+		return FALSE
+	var/list/kudzu_in_area = range(2, src)
 
-/mob/living/simple_animal/hostile/venus_human_trap/AttackingTarget()
-	. = ..()
-	if(isliving(target))
-		var/mob/living/L = target
-		if(L.stat != DEAD)
-			adjustHealth(-maxHealth * 0.1)
-
-/mob/living/simple_animal/hostile/venus_human_trap/OpenFire(atom/the_target)
-	for(var/datum/beam/B in vines)
-		if(B.target == the_target)
-			pull_vines()
-			ranged_cooldown = world.time + (ranged_cooldown_time * 0.5)
-			return
-	if(get_dist(src,the_target) > vine_grab_distance || vines.len >= max_vines)
+	if(!(locate(/obj/structure/spacevine) in kudzu_in_area))
+		apply_damage(10, BRUTE, BODY_ZONE_CHEST) //tick 5 times to die
+		balloon_alert(src, "do not leave vines!")
+
+/datum/action/cooldown/mob_cooldown/vine_tangle
+	name = "Tangle"
+	button_icon = 'icons/mob/spacevines.dmi'
+	button_icon_state = "Light1"
+	desc = "Trip and pull things."
+	cooldown_time = 8 SECONDS
+	///how many vines can we handle
+	var/max_vines = 2
+	/// An assoc list of all the plant's vines (beam = leash)
+	var/list/vines = list()
+	/// How far away a plant can attach a vine to something
+	var/vine_grab_distance = 4
+	/// how long does a vine attached to something last (and its leash)
+	var/vine_duration = 1.5 SECONDS
+
+/datum/action/cooldown/mob_cooldown/vine_tangle/Remove(mob/remove_from)
+	for(var/datum/beam/vine as anything in vines)
+		qdel(vine)
+	return ..()
+
+/datum/action/cooldown/mob_cooldown/vine_tangle/Activate(atom/target_atom)
+	if(isturf(target_atom))
+		return
+	if(get_dist(owner, target_atom) > vine_grab_distance || vines.len >= max_vines)
 		return
-	for(var/turf/T in get_line(src,target))
-		if (T.density)
+	for(var/turf/blockage in get_line(owner, target_atom))
+		if (blockage.density)
 			return
-		for(var/obj/O in T)
-			if(O.density)
+		for(var/obj/blockade in blockage)
+			if(blockade.density)
 				return
 
-	var/datum/beam/newVine = Beam(the_target, icon_state = "vine", maxdistance = vine_grab_distance, beam_type=/obj/effect/ebeam/vine, emissive = FALSE)
+	var/datum/beam/newVine = owner.Beam(target_atom, icon_state = "vine", time = vine_duration, beam_type=/obj/effect/ebeam/vine, emissive = FALSE)
+	var/component = target_atom.AddComponent(/datum/component/leash, owner, vine_grab_distance)
```

Comment:
Is this a change in function? I think this would previously pull people closer, while this only prevents them from leaving.

Reply:
you can move backwards to pull them lasso style

Reply:
Previously it would just automatically reel you in, no? This is a pretty significant change in function which you did not document and it isn't clear it was even intentional.

Reply:
the previous ""function"" didnt even work to begin with

Reply:
Fix it then

Reply:
what if i write it in the changelog

Reply:
That's the minimum requirement but I will probably still disapprove of changing it in this way

Reply:
For reference, this did work at one point, but it broke not all that long ago.